src/api.cpp

```diff
@@ -2893,7 +2894,12 @@ cl_int CLVK_API_CALL clGetKernelInfo(cl_kernel kern, cl_kernel_info param_name,
         copy_ptr = &val_program;
         ret_size = sizeof(val_program);
         break;
-    case CL_KERNEL_ATTRIBUTES: // TODO implement
+    case CL_KERNEL_ATTRIBUTES: {
+        const auto& attrs = kernel->attributes();
+        copy_ptr = attrs.c_str();
+        ret_size = attrs.size() + 1;
```

Comment:
You could use `api_query_string` to avoid having to add the + 1 manually but this would mean a copy unless we use it in the objects directly. I think the copy is acceptable for this query.

Reply:
Done